client/src/java/com/zimbra/client/ZMailbox.java

```diff
@@ -2953,9 +2953,16 @@ public ZFolder getFolderById(String id) throws ServiceException {
             /* sometimes when working on behalf of, we're getting just by item id but the
              * cache has fully qualified keys.
              */
-            String ident = new ItemIdentifier(id, this.getAccountId()).toString();
-            if (!id.equals(ident)) {
-                item = mItemCache.getById(ident);
+            try {
+                String ident = new ItemIdentifier(id, this.getAccountId()).toString();
+                if (!id.equals(ident)) {
+                    item = mItemCache.getById(ident);
+                }
+            } catch (ServiceException e) {
+                // This can be raised when constructing an ItemIdentifier with a folder name; e.g., "INBOX"
+                if (!(e.getCause() instanceof NumberFormatException)) {
+                    throw e;
```

Comment:
so if a caller passes "INBOX" instead of an number, the caller gets NULL. Does this mean that the caller is misusing this method or that INBOX does not exist?